docs/advanced/shebang.md

```diff
@@ -0,0 +1,109 @@
+---
+part: pixi/advanced
+title: Self-contained scripts
+description: Learn how to write self-contained shell scripts using `pixi exec`
+---
+
+# Using `pixi exec` to create self-contained scripts
+
+!!!warning "Only on Unix-like systems"
+    The following approach only works on Unix-like systems (i.e. Linux and macOS) since Windows does not support shebang lines.
+
+For simple scripts, you can use [`pixi exec`](../reference/cli.md#exec) to run them directly without needing to take care of installing dependencies or setting up a virtual environment.
+This can be done by adding a [shebang line](https://en.wikipedia.org/wiki/Shebang_(Unix)) at the top of the script, which tells the system how to execute the script.
+Usually, a shebang line starts with `#!/usr/bin/env` followed by the name of the interpreter to use.
+
+Instead of adding an interpreter, you can also just add `pixi exec` at the beginning of the script.
+The only requirement for your script is that you must have `pixi` installed on your system.
+
+!!!tip "Making the script executable"
+    You might need to make the script executable by running `chmod +x my-script.sh`.
+
+```bash title="deploy-tf.sh"
+#!/usr/bin/env -S pixi exec --spec opentofu -- bash -e
```

Comment:
Would be nice to create a less specific example, might be as simple as just running `python` or `bat`

Reply:
```bash title="use-bat.sh"
#!/usr/bin/env -S pixi exec --spec bat -- bash -e

bat my-file.json
```

feels a bit artificially crafted though but i guess it should be fine

Reply:
[`45a979e` (#3201)](https://github.com/prefix-dev/pixi/pull/3201/commits/45a979eb82674999daa9083ae916d2a9edfb44eb)